core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/DataAccessImpl.java

```diff
@@ -522,6 +524,13 @@ protected void initPreparedStatements() {
             "FROM metrics_idx " +
             "WHERE tenant_id = ? AND type = ? AND metric = ?");
 
+        scanMetricInMetricsIndex = session.prepare(
+                "SELECT tenant_id, type, metric, tags, data_retention, token(tenant_id, type) " +
```

Comment:
Unless heapster does something with the `data_retention` column, I would remove it from the select clause.

Reply:
Yeah, I made it so that the Metric type is full, if someone would suddenly need it.. Cassandra has to read it in any case as it can't jump past it.

Reply:
It has to read it, but if removed from the select clause, will the cell still get deserialized onto the heap and then sent across the wire for each row? I assume no.